### PR TITLE
Removing termController and adding a filter by letter

### PR DESCRIPTION
**Pull request by andrewmp1**
I've merged your work on removing the termController and I added another feature by where we can filter by the alphabet list on the right now.

### CODE REVIEW

File: app/assets/javascripts/app.js.coffee [hunk 1]
```diff
@@ -4,33 +4,47 @@ exports.YJ = Em.Application.create()
...
+
+YJ.termsController = Em.ArrayProxy.create(
+  content: []
   currentTerm: null
+  searchLetter: null
 
   newTerm: ->
-    term = YJ.Term.create(term: "A term", definition: "A description")
-    @editTerm(term)
+    @editTerm(YJ.Term.create(term: "A term", definition: "A description"))
```
Comment (oldfartdeveloper) [inline app/assets/javascripts/app.js.coffee:24]: We need to introduce a new term the way the "todos" project does it.  Hints in the text edit boxes.

File: app/assets/javascripts/app.js.coffee [hunk 2]
```diff
@@ -4,33 +4,47 @@ exports.YJ = Em.Application.create()
...
     YJ.editTermView.remove()
     $("#indexTermView").show()
 
-)
+  filtered: (->
+    if @get("searchLetter") is null
+      @get('content')
+    else
+      filteredList = @get('content').filterProperty 'firstLetter', @get('searchLetter')
+  ).property('searchLetter').cacheable()
```
Comment (oldfartdeveloper) [inline app/assets/javascripts/app.js.coffee:46]: Cool!

File: spec/javascripts/filter-terms-spec.js.coffee [hunk 3]
```diff
@@ -0,0 +1,14 @@
+describe "Filtered terms", ->
+
+  beforeEach ->
+  @ctl = YJ.termsController
+  @ctl.set('content', [])
+  @ctl.load()
+
+  it "matches the first letter of the term to the submitted letter", ->
```
Comment (oldfartdeveloper) [inline spec/javascripts/filter-terms-spec.js.coffee:8]: Any reason this test is commented out?

**Comment by oldfartdeveloper** [comment]
Thanks for the new feature!

File: spec/javascripts/filter-terms-spec.js.coffee (hunk 3 above)
Comment (andrewmp1) [inline spec/javascripts/filter-terms-spec.js.coffee:8]: Ya,

I was having a real pain trying to get it to pass.  If I added a beforeEach to set up the test it started breaking all the rest of the tests so I just commented it out and thought maybe we could take a look at it together.

Comment (oldfartdeveloper) [inline spec/javascripts/filter-terms-spec.js.coffee:8]: Doncha love it?  How's your schedule looking for Tuesday night?  I'll be available

Comment (oldfartdeveloper) [inline spec/javascripts/filter-terms-spec.js.coffee:8]: How's your schedule look for Tuesday night?  I'll be available.

**Pull request merged**